plugins/osdn/node.go

```diff
@@ -0,0 +1,140 @@
+package osdn
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/golang/glog"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	osapi "github.com/openshift/origin/pkg/sdn/api"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/container"
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+	kubeutilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+type OsdnNode struct {
+	multitenant        bool
+	registry           *Registry
+	localIP            string
+	localSubnet        *osapi.HostSubnet
+	hostName           string
+	podNetworkReady    chan struct{}
+	vnids              vnidMap
+	iptablesSyncPeriod time.Duration
+	mtu                uint
+}
+
+// Called by higher layers to create the plugin SDN node instance
+func NewNodePlugin(pluginName string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string, iptablesSyncPeriod time.Duration, mtu uint) (api.OsdnNodePlugin, error) {
```

Comment:
We have NewNodePlugin(), NewProxyPlugin() but for master it is StartMaster()..can we reintroduce NewMasterPlugin() to be consistent?